app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1382,9 +1416,20 @@ class BrowserTabViewModel(
         }
     }
 
-    suspend fun onBookmarkAddRequested() {
+    fun onBookmarkMenuClicked() {
         val url = url ?: return
-        val title = title ?: ""
+        viewModelScope.launch {
+            if (currentBrowserViewState().bookmark != null) {
+                pixel.fire(AppPixelName.MENU_ACTION_EDIT_BOOKMARK_PRESSED.pixelName)
+                editSiteBookmark(url)
```

Comment:
See my other comment. If we can obtain the bookmark at this point, we could replace `editSiteBookmark` for `onEditSavedSiteRequested`.